src/AbstractOperations/binary_operations.jl

```diff
@@ -187,6 +187,36 @@ macro binary(ops...)
     return expr
 end
 
+#####
+##### BinaryOperations with ZeroField
+#####
+
+import Base: +, -, *, /, ==
+using Oceananigans.Fields: ZeroField
+
+==(::ZeroField, ::ZeroField) = true
+
++(a::ZeroField, b::AbstractField) = b
++(a::AbstractField, b::ZeroField) = +(b, a)
++(a::ZeroField, b::Number) = b
++(a::Number, b::ZeroField) = +(b, a)
+
+-(a::ZeroField, b::AbstractField) = -b
+-(b::AbstractField, a::ZeroField) = b
+-(a::ZeroField, b::Number) = -b
```

Comment:
And maybe this `ConstantField(-b)` and so on?

Reply:
Ok, sorry I missed this comment. I gave the opposite advice below so we should discuss.